rust/src/http2/http2.rs

```diff
@@ -146,6 +149,12 @@ pub struct HTTP2Transaction {
     pub escaped: Vec<Vec<u8>>,
     pub req_line: Vec<u8>,
     pub resp_line: Vec<u8>,
+
+    is_doh_response: bool,
```

Comment:
should this all be wrapped in an Option? It seems like a large expansion of the http/2 tx

Reply:
Looks like a good idea, will try to improve on it